merkletree/str.go

```diff
@@ -30,6 +31,41 @@ type SignedTreeRoot struct {
 	Ad              AssocData
 }
 
```

Comment:
Maybe add here,

// TODO(arlolra): Look into using a `DirSTR` instead, to avoid this mess.